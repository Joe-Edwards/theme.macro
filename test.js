const path = require('path');
const pluginTester = require('babel-plugin-tester').default;
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  babelOptions: { filename: __filename },
  tests: [
    { fixture: path.join(__dirname, 'tests', 'valid.js'), snapshot: true, },
    { fixture: path.join(__dirname, 'tests', 'error.js'), error: 'The theme macro at line 4 is not used within a template expression' }
  ],
});
