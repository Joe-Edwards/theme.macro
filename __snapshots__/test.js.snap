// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `

import theme from '../theme.macro';

\`\${theme.foo}\` // Simple member access
\`\${theme.foo.bar}\` // Nested member access
\`\${theme.foo[8] + 7}\` // Complex expression
\`\${props => theme.foo + props.bar}\` // Expression with props
\`\${theme.foo + theme.bar}\` // Expression with multiple references

      ↓ ↓ ↓ ↓ ↓ ↓

\`\${(_props) => _props.theme.foo}\` // Simple member access
\`\${(_props2) => _props2.theme.foo.bar}\` // Nested member access
\`\${(_props3) => _props3.theme.foo[8] + 7}\` // Complex expression
\`\${(_props4) => (props) => _props4.theme.foo + props.bar}\` // Expression with props
\`\${(_props5) => _props5.theme.foo + _props5.theme.bar}\`; // Expression with multiple references


`;
